src/main/java/DukeTask/Task.java

```diff
@@ -0,0 +1,133 @@
+ * by the end-user.
+ */
+public abstract class Task {
+    /**
+     * Exception thrown when the user does not provide any
+     * task description on construction.
+     */
+    public static class EmptyDescriptionException extends Exception {
+        private static final long serialVersionUID = -365632185953600895L;
+
+        public EmptyDescriptionException() {
+            super("task description cannot be empty! no sense...");
+        }
+    }
+
+    /**
+     * Exception thrown when user attempts to mark the task as done
+     * when the task is already in a done state.
+     */
+    public static class MarkedAsDoneException extends Exception {
+        private static final long serialVersionUID = -7397628460804553793L;
+
+        public MarkedAsDoneException() {
+            super("trickery! task has already been marked done by dogeDuke...");
+        }
+    }
+
+    /**
+     * Enum switch state for whether the task is done or not.
+     */
+    protected enum TaskState {
+        done, undone;
+    }
+
+    protected String description;
+    protected TaskState state;
+    protected LocalDateTime createdDateTime;
+
+    /**
+     * Class main constructor.
+     * @param description       the task description; cannot be empty.
+     * @param isDone            whether the task is done on creation.
+     * @param createdDateTime   the datetime at which the task is created.
+     * @throws Task.EmptyDescriptionException
+     */
+    public Task (String description, boolean isDone, LocalDateTime createdDateTime)
+            throws Task.EmptyDescriptionException {
+
+        if (description.isEmpty()) {
+            throw new Task.EmptyDescriptionException();
+        }
+
+        this.state = (isDone ? TaskState.done : TaskState.undone);
+        this.description = description;
+        this.createdDateTime = createdDateTime;
+    }
+
+    /**
+     * Class default constructor.
+     * passes the undone state and current datetime for the task isDone and
+     * createdDateTime parameters.
+     * @param description       the task description; cannot be empty.
+     * @throws Task.EmptyDescriptionException
+     */
+    public Task (String description) throws Task.EmptyDescriptionException {
+        this(description, false, LocalDateTime.now());
+    }
+
+    // accessors
+    public boolean isDone () {
+        return (this.state == TaskState.done);
+    }
+
+    public boolean inDescription (String keyphrase) {
+        return this.description.indexOf(keyphrase) >= 0;
+    }
+
+    /**
+     * Returns information about the task.
+     * @param outputFormat  the format in which to output any datetime
+     *                      attributes as string representation.
+     * @return      string containing the current state, the description
+     *              and the datetime of creation in the format of the
+     *              specified outputFormat.
+     */
+    public String taskInformation (DateTimeFormatter outputFormat) {
```

Comment:
According to Java coding standard, `Names representing methods must be verbs ...`. Maybe a good alternative for this method name would be `getTaskInformation`?